### ISSUE TITLE
rightclicking on a block in the editor can close the context menu in the same click

### ISSUE THREAD

**Issue by peabrainiac** [LLK/scratch-gui#5888]
### Expected Behavior

When opening a menu by rightclicking on a block in the editor, the menu shouldn't be able to close again because of that same click.

### Actual Behavior

When moving your mouse while rightclicking, you may sometimes immediately select one of the options with the same rightclick you opened them with.

### Steps to Reproduce

Right click on a block, move your mouse over one of the options and then let go of the right mouse button again.

Or, just rightclick while moving your mouse down and to the right.

### Operating System and Browser

Windows 10, Firefox 77 and Chrome 83

**Comment by peabrainiac** [comment]
I don't know the actual internals at play here, but to me it looks like there are two easy ways to solve this problem:

One would be to open the menu when the mouse button is released instead of when it's pressed - this would also be consistent with how Firefox and Chrome usually handle right clicks.

The other would be to create a one-time event handler that cancels the next mouseup event so no other action can take place when the button is released again.

**Comment by BryceLTaylor** [comment]
This appears to happen on all browsers on Windows only.

**Comment by peabrainiac** [comment]
I'd like to try to create a pull request that fixes this, but I'm a little confused as to whether this is the correct repository to do so - are these menus part of scratch-gui or of scratch-blocks?

**Comment by chrisgarrity** [comment]
@peabrainiac these menus are part of scratch-blocks. So I'll move it to the scratch-blocks repository. This will not get the 'help wanted' label because it will be difficult to test - it will need to work on all browser/OS version and on touch devices without a mouse.

**Comment by peabrainiac** [comment]
yup. I can't get scratch-blocks to work on my machine anyways, because of [#2080](https://github.com/LLK/scratch-blocks/issues/2080) - it's probably better if someone else does it.
